lib/archethic/beacon_chain.ex

```diff
@@ -338,12 +354,50 @@ defmodule Archethic.BeaconChain do
   """
   @spec list_transactions_summaries_from_current_slot(DateTime.t()) ::
           list(TransactionSummary.t())
-  def list_transactions_summaries_from_current_slot(date = %DateTime{} \\ DateTime.utc_now()) do
-    next_summary_date = next_summary_date(DateTime.truncate(date, :millisecond))
+  def list_transactions_summaries_from_current_slot(datetime = %DateTime{} \\ DateTime.utc_now()) do
+    get_next_summary_elected_subsets_by_nodes(datetime)
+    |> Task.async_stream(
+      fn {node, subsets} ->
+        fetch_current_summaries(node, subsets)
+      end,
+      ordered: false,
+      max_concurrency: 256
+    )
+    |> Stream.filter(&match?({:ok, _}, &1))
+    |> Stream.flat_map(fn {:ok, summaries} -> summaries end)
+
+    # remove duplicates & sort
+    |> Stream.uniq_by(& &1.address)
+    |> Enum.sort_by(& &1.timestamp, {:desc, DateTime})
+  end
 
+  @doc """
+  Return the replications attestation that were gathered since last
+  Unordered
+  """
+  @spec fetch_current_summary_replication_attestations(datetime :: DateTime.t()) ::
+          list(ReplicationAttestation.t())
+  def fetch_current_summary_replication_attestations(datetime = %DateTime{} \\ DateTime.utc_now()) do
+    get_next_summary_elected_subsets_by_nodes(datetime)
+    |> Task.async_stream(
+      fn {node, subsets} ->
+        {:ok, replications_attestations} =
```

Comment:
Here you do not handle the `:error` result. So if it happens, the bootstrap will crash.
Maybe you can do the same as the transaction summaries, return an empty list if there is an error